certmanager/ca.go

```diff
@@ -14,7 +14,71 @@
 
 package certmanager
 
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"fmt"
+	"time"
+)
+
+const (
+	// The time to live for issued certificate.
+	certTTL = time.Hour
+	// The time to live for self-signed root CA certificate.
+	rootCertTTL = 24 * time.Hour
+)
+
 // CertificateAuthority contains methods to be supported by a CA.
 type CertificateAuthority interface {
-	Generate(name string) (key, cert []byte)
+	Generate(name, namespace string) (cert, key []byte)
+}
+
+// IstioCA generates keys and certificates for Istio identities.
+type IstioCA struct {
+	signerCert *x509.Certificate
+	signerKey  *rsa.PrivateKey
```

Comment:
use `crypto.PrivateKey` so the key doesn't have to be RSA (it could be ECDSA)? or we can just mark it for now we only support RSA keys.